components/camel-bindy/src/main/java/org/apache/camel/dataformat/bindy/csv/BindyCsvDataFormat.java

```diff
@@ -138,58 +142,74 @@ public Object unmarshal(Exchange exchange, InputStream inputStream) throws Excep
         // List of Pojos
         List<Map<String, Object>> models = new ArrayList<>();
 
-        // Pojos of the model
-        Map<String, Object> model;
         InputStreamReader in = null;
-        Scanner scanner = null;
         try {
             if (checkEmptyStream(factory, inputStream)) {
                 return models;
             }
     
             in = new InputStreamReader(inputStream, IOHelper.getCharsetName(exchange));
-    
-            // Scanner is used to read big file
-            scanner = new Scanner(in);
-    
+
             // Retrieve the separator defined to split the record
             String separator = factory.getSeparator();
             String quote = factory.getQuote();
             ObjectHelper.notNull(separator, "The separator has not been defined in the annotation @CsvRecord or not instantiated during initModel.");
-    
-            int count = 0;
-            
-            // If the first line of the CSV file contains columns name, then we
-            // skip this line
-            if (factory.getSkipFirstLine()) {
-                // Check if scanner is empty
-                if (scanner.hasNextLine()) {
-                    scanner.nextLine();
+            AtomicInteger count = new AtomicInteger(0);
+
+            // Use a Stream to stream a file across.
+            try (Stream<String> lines = new BufferedReader(in).lines()) {
+                int linesToSkip = 0;
+
+                // If the first line of the CSV file contains columns name, then we
+                // skip this line
+                if (factory.getSkipFirstLine()) {
+                    linesToSkip = 1;
                 }
-            }
-    
-            while (scanner.hasNextLine()) {
-    
-                // Read the line
-                String line = scanner.nextLine().trim();
-    
-                if (ObjectHelper.isEmpty(line)) {
-                    // skip if line is empty
-                    continue;
+
+                // Consume the lines in the file via a consumer method, passing in state as necessary.
+                // If the internals of the consumer fail, we unrap the checked exception upstream.
+                try {
+                    lines.skip(linesToSkip)
+                            .forEachOrdered(consumeFile(factory, models, separator, quote, count));
+                } catch (WrappedException e) {
+                    throw e.getWrappedException();
```

Comment:
i did not get the need of WrappedException

Reply:
Because we're performing this operation inside of a `Stream`, we have to throw unchecked exceptions.  Streams do not allow us to propagate a checked exception, and the operations it's doing have the potential to throw checked exceptions, so wrapping it in an unchecked exception (and later throwing it if necessary) is the way to work around that limitation in `Stream`s.